### PR TITLE
Fix bidirectional mode in reference implementations of GRU/LSTM/RNN Sequences

### PR DESCRIPTION
**Pull request by itikhono**
Changes:

- Fix reference implementations
- Enable decomposition of bidirectional Sequences as part of CommonOptimizations (ticket 39222)
- Update single layer tests

### CODE REVIEW

**Review by iefode — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp [hunk 1]
```diff
@@ -53,6 +54,9 @@ bool ngraph::pass::CommonOptimizations::run_on_function(std::shared_ptr<ngraph::
     manager.register_pass<ngraph::pass::HSwishFusion>();
     manager.register_pass<ngraph::pass::ConvertPadToGroupConvolution>();
     manager.register_pass<ngraph::pass::NormalizeL2Fusion>();
+    manager.register_pass<ngraph::pass::BidirectionalLSTMSequenceDecomposition>();
+    manager.register_pass<ngraph::pass::BidirectionalRNNSequenceDecomposition>();
+    manager.register_pass<ngraph::pass::BidirectionalGRUSequenceDecomposition>();
```
Comment (ilyachur) [inline inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp:59]: Do we need to do those decompositions for all plugins?

Comment (itikhono) [reply inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp:59]: I will clarify this point.

Comment (itikhono) [reply inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp:59]: cpu/gpu/vpu/onednn plugins don't support Bidirectional mode yet. 
As far as I know, there are no plans to add such support anytime soon.
Anyway, the transformations can be disabled using callbacks.

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Do we have tests for reference implementation?

**Comment by itikhono** [comment]
> Do we have tests for reference implementation?

Yes, we have. There are single-layer tests for gru/lstm/rnn cell and sequences and some onnx importer and ngraph unit tests.

**Review by ilyachur — APPROVED** [review]

**Pull request merged**